docs/references/ic-interface-spec.md

```diff
@@ -1944,6 +1945,26 @@ The argument `type` decides which performance counter to return:
 
 In the future, the IC might expose more performance counters.
 
+### Canister limit {#system-api-canister-limit}
+
+The canister can query one of the "canister limits", which denotes a resource limit of the canister imposed by the current configuration of the IC and/or canister. This allows runtime logic, such garbage collector or persistence mechanism, to adapt to the system limits without recompilation and redeployment.
+
+`ic0.canister_limit : (limit_type : i32) -> i64`
+
+The argument `limit_type` determines which resource limit to return:
```

Comment:
Should we add a limit for the remaining memory (heap or stable) pages that could still be allocated by the cansiter>